hub/codec/codecs.py

```diff
@@ -0,0 +1,380 @@
+from io import BytesIO
+import numcodecs
+from numcodecs.abc import Codec
+
+import numpy as np
+from PIL import Image
+
+
+class Base:
+    """Base class for numcodec compressors"""
+
+    def encode(self, array: np.ndarray) -> bytes:
```

Comment:
can we call these methods "compress" & "decompress" instead? it would be easier to read IMO

Reply:
for lz4 it would be `bytes`, for png i think it would be `np.array`

Reply:
actually, maybe you should just defer this typing to the class that inherits from this base.

Reply:
I think it will be more intuitive to call the method by the name of the main function used in it. And in png and jpeg cases we just overwrite `encode` and `decode` methods from Codec.